From 9732c8b85c4347f2c807f287e07288da62057808 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <quic_jouni@quicinc.com>
Date: Thu, 25 Jul 2024 00:39:30 +0000
Subject: [PATCH] wlantest: Remove unused QoS handling from GCMP nonce
 generation

This was copied from CCMP, but is not needed (and can trigger compiler
warnings) with GCMP.

Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 wlantest/gcmp.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/wlantest/gcmp.c b/wlantest/gcmp.c
index a29496d30..f0bcec2ba 100644
--- a/wlantest/gcmp.c
+++ b/wlantest/gcmp.c
@@ -32,12 +32,8 @@ static void gcmp_aad_nonce(const struct ieee80211_hdr *hdr, const u8 *data,
 	if (WLAN_FC_GET_TYPE(fc) == WLAN_FC_TYPE_DATA) {
 		fc &= ~0x0070; /* Mask subtype bits */
 		if (stype & 0x08) {
-			const u8 *qc;
 			qos = 1;
 			fc &= ~WLAN_FC_HTC;
-			qc = (const u8 *) (hdr + 1);
-			if (addr4)
-				qc += ETH_ALEN;
 		}
 	}
 
-- 
2.43.0

